ext/openssl/openssl.c

```diff
@@ -4786,7 +4787,8 @@ PHP_FUNCTION(openssl_pkey_derive)
 		RETURN_THROWS();
 	}
 	if (key_len < 0) {
-		php_error_docref(NULL, E_WARNING, "keylen < 0, assuming NULL");
+		zend_argument_value_error(3, "must be greater than or equal to 0");
```

Comment:
this will be inconsistent with other checks - please change to warning.

Reply:
Ok you can use actually error in here as it's a programming error so ignore my comment